### PR TITLE
Bootstrap chatbot user & mediator

### PR DESCRIPTION
**Pull request by euanmillar**

### CODE REVIEW

File: infrastructure/openhim-base-config.json [hunk 1]
```diff
@@ -1524,6 +1524,57 @@
...
+          "primary": true,
+          "username": "",
+          "password": ""
+        }
+      ],
+      "requestBody": true,
+      "responseBody": true,
+      "rewriteUrlsConfig": [],
+      "name": "Chatbot mediator",
+      "urlPattern": "^/search",
```
Comment (mushrafulhoque-dsi) [inline infrastructure/openhim-base-config.json]: Should we try to give a more specific url pattern for chatbot search feature ? something like: ` ^/chat-bot/search`

File: packages/user-mgnt/src/utils/userUtils.ts [hunk 2]
```diff
@@ -30,7 +30,8 @@ export const roleScopeMapping: IRoleScopeMapping = {
   PERFORMANCE_OVERSIGHT: ['performance'],
   PERFORMANCE_MANAGEMENT: ['performance'],
   NOTIFICATION_API_USER: ['declare', 'notification-api'],
-  VALIDATOR_API_USER: ['validator-api']
+  VALIDATOR_API_USER: ['validator-api'],
+  CHATBOT_API_USER: ['declare', 'chatbot-api']
```
Comment (mushrafulhoque-dsi) [inline packages/user-mgnt/src/utils/userUtils.ts]: does it need to be on `declare` scope as well ?

Comment (euanmillar) [reply packages/user-mgnt/src/utils/userUtils.ts]: Yes, it needs either the declare or register scope in order to access the search

File: infrastructure/openhim-base-config.json (hunk 1 above)
Comment (euanmillar) [reply infrastructure/openhim-base-config.json]: I think that we should.  But I actually think we should have a dedicated endpoint for chatbot search that doesn't return all the data points that the search endpoint does for security reasons.  Right now though, this is a prototype that is useless without the chatbot mediator and chatbot application.  So for now I think this is fine.

**Pull request merged**